tendrl/commons/tests/jobs/test_jobs_init.py

```diff
@@ -1,259 +1,126 @@
-            global test_job
-            NS.type = "Test_type"
-            NS.node_context.tags = "No_tag"
-            test_job.run()
-            test_job._complete._Event__flag = False
-            NS.node_context.tags = "Test_tag"
-            test_job.run()
-            test_job._complete._Event__flag = False
-            mock_load.return_value = load("node")
-            NS.node_context.node_id = "Test_node"
-            test_job.run()
-            test_job._complete._Event__flag = False
-            mock_load.return_value = load("no_obj_name")
-            test_job.run()
-            test_job._complete._Event__flag = False
-            with patch.object(Client, "write", write):
-                test_job.run()
-                test_job._complete._Event__flag = False
-            with patch.object(Client, "write", status_write):
-                test_job.run()
-                test_job._complete._Event__flag = False
-            with patch.object(Client, "write", failed_write):
-                with patch.object(NameSpace, "run", run):
-                    test_job.run()
-                    test_job._complete._Event__flag = False
-            with patch.object(Client, "write", failed_write("no_err")):
-                with patch.object(NameSpace, "run", run):
-                    test_job.run()
-                    test_job._complete._Event__flag = False
-    test_job = JobConsumerThread()
-    test_job._complete._Event__flag = True
-    test_job.run()
-    NS.node_context.tags = "tendrl/monitor"
-    test_job._complete._Event__flag = True
-    test_job.run()
-
-
-def test_stop():
-    test_job = JobConsumerThread()
-    test_job.stop()
-    assert test_job._complete._Event__flag
+@patch.object(threading, "Thread")
+@patch.object(time, "sleep")
+@patch.object(etcd_utils, "read", read)
+@patch.object(BaseObject, "load_all")
+def test_job_consumer_thread(load_all, sleep, thread):
+    test_init.init()
+    thread.return_value = MagicMock()
+    sleep.return_value = None
+    job = NS.tendrl.objects.Job(
+        job_id="808a4162-4b70-4ff0-b218-45dbe371e545",
+        status="new",
+        payload={"status": "new"}
+    )
+    load_all.return_value = [job]
+    NS.publisher_id = "testing"
+    obj = jobs.JobConsumerThread()
+    with patch.object(NS.node_context, "load") as nc_load:
+        NS.node_context.fqdn = "tendrl-node-1"
+        nc_load.return_value = NS.node_context
+        with patch.object(NS.tendrl_context, "load") as tc_load:
+            with patch.object(obj._complete, "is_set", is_set):
+                NS.tendrl_context.integration_id = \
+                    "b0b18359-3444-40b7-aa99-853f1b7308de"
+                tc_load.return_value = NS.tendrl_context
+                obj.run()
+
+
+@patch.object(logger, "log")
+def test_process_job_fail(log):
+    # no tag match
+    test_init.init()
+    NS.type = "node"
+    NS.publisher_id = "pytest"
+    NS.tendrl.objects.Job.load = MagicMock()
```

Comment:
Should we use MagicMock @r0h4n ?

Reply:
same case here 

using mock/patch we can mock the function of a class which is existing one, but here I have created a mock function for a class in the name load.

Because TendrlContext class does not have load function, only BaseObject class have load function, when we create an object for TendrlContext then only base class (Baseobject) properties should be accessible by child class Obj. But here I want to create one mock function for class TendrlContext before object creation. So child class method overrides base class.

Reply:
ack